### PR TITLE
[wip] UI

### PR DESCRIPTION
**Pull request by cassiozen**
**- Summary**

- Infer title, image and list all other fields
- Fixed size (remove masonry layout)

**- A picture of a cute animal (not mandatory but encouraged)**
```
         ,--._______,-. 
       ,','  ,    .  ,_`-. 
      / /  ,' , _` ``. |  )       `-.. 
     (,';'""`/ '"`-._ ` \/ ______    \\ 
       : ,o.-`- ,o.  )\` -'      `---.)) 
       : , d8b ^-.   '|   `.      `    `. 
       |/ __:_     `. |  ,  `       `    \ 
       | ( ,-.`-.    ;'  ;   `       :    ; 
       | |  ,   `.      /     ;      :    \ 
       ;-'`:::._,`.__),'             :     ; 
      / ,  `-   `--                  ;     | 
     /  \                   `       ,      | 
    (    `     :              :    ,\      | 
     \   `.    :     :        :  ,'  \    : 
      \    `|-- `     \ ,'    ,-'     :-.-'; 
      :     |`--.______;     |        :    : 
       :    /           |    |         |   \ 
       |    ;           ;    ;        /     ; 
     _/--' |           :`-- /         \_:_:_| 
   ,',','  |           |___ \ 
   `^._,--'           / , , .) 
                      `-._,-' 
```

### CODE REVIEW

**Comment by biilmann** [comment]
This works much better!

Still in doubt if we'll need a traditional list/table based UI as well, but now the card UI works so well for some types of collections (like the Smashing Author's collection) that once we get it polished design wise, I believe it'll be the best solution in some cases.

Currently the Card UI field inference breaks with file based collections (like the settings collection in the example page).

The Dashboard needs something more that the editorial flow. Right now for a new CMS install it's simply an empty page.

File: src/components/Card.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+export default class ImageCard extends React.Component {
+
+  render() {
+    const { onClick, onImageLoaded, image, text, description } = this.props;
+    return (
+      <Card onClick={onClick} className={styles.root}>
+        <img src={image} onLoad={onImageLoaded} />
+        <h1>{text}</h1>
+
+        {description ? <p>{description}</p> : null}
```
Comment (okonet) [inline src/components/Card.js]: Let's rewrite this as 

```
{ description && <p>{ description }</p> }
```

File: src/reducers/collections.js [hunk 2]
```diff
@@ -14,6 +15,8 @@ const collections = (state = null, action) => {
             configCollection.type = FOLDER; // eslint-disable-line no-param-reassign
           } else if (hasProperty(configCollection, 'files')) {
             configCollection.type = FILES; // eslint-disable-line no-param-reassign
+          } else {
+            throw ('Unknown collection type. Collections can be either Folder based or File based. Please verify your site configuration');
```
Comment (okonet) [inline src/reducers/collections.js]: Shouldn't it be `throw new Error()`? We have a lint rule for that AFAIK.

File: src/components/Card.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React, { PropTypes } from 'react';
+import { Card } from '../UI';
+import styles from './ImageCard.css';
```
Comment (okonet) [inline src/components/Card.js]: This can't be resolved since the path is wrong.

File: src/components/Card.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  image: PropTypes.string,
+  onClick: PropTypes.func,
+  onImageLoaded: PropTypes.func,
+  text: PropTypes.string.isRequired,
+  description: PropTypes.string,
+};
+
+ImageCard.defaultProps = {
+  onClick() {},
+  onImageLoaded() {},
```
Comment (okonet) [inline src/components/Card.js]: Not needed anymore?

File: src/components/Card.js (hunk 3 above)
Comment (okonet) [inline src/components/Card.js]: This can't be resolved since the path is wrong.

**Comment by cassiozen** [comment]
@biilmann Yes, I'm still working on both the file based collections introduced bug and the empty dashboard (for those with no editorial workflow.) In the last case, I think I'll simply redirect to the first collection (as before)

File: src/reducers/collections.js (hunk 2 above)
Comment (cassiozen) [inline src/reducers/collections.js]: Yes, thanks. Fixed.

File: src/components/Card.js (hunk 3 above)
Comment (cassiozen) [inline src/components/Card.js]: Unused file was lingering around. Deleted it.

File: src/components/ControlPanel/ControlPane.js [hunk 5]
```diff
@@ -38,7 +42,7 @@ export default class ControlPane extends Component {
       <div>
         {
           fields.map(field =>
-            <div
+            isHidden(field) ? null : <div
```
Comment (okonet) [inline src/components/ControlPanel/ControlPane.js]: `!isHidden(field) && <div>`?

**Pull request merged**